examples/allsmt.py

```diff
@@ -0,0 +1,33 @@
+# Perform ALL-SMT on Theory atoms
+#
+# This example shows:
+#  - How to get the logic of a formula
+#  - How to enumerate models
+#  - How to extract a partial model
+#
+from pysmt.shortcuts import Solver, Equals, Not, And, Symbol
+from pysmt.shortcuts import LE, GE, Int, Plus
+from pysmt.typing import INT
+from pysmt.oracles import get_logic
+
+def all_smt(formula, keys):
+    target_logic = get_logic(formula)
+    print("Target Logic: %s" % target_logic)
+    with Solver(logic=target_logic) as solver:
+        solver.add_assertion(formula)
+        while solver.solve():
+            partial_model = [Equals(k, solver.get_value(k)) for k in keys]
```

Comment:
In order to be complete, check the type of k and if it is Bool use Iff, this better shows the difference between pysmt and z3py